### PR TITLE
785 unit test tasks

### PR DESCRIPTION
**Pull request by dustinleblanc**
Easiest test PR ever.

This module IMO could use some improvements. It doesn't validate the task object from what I can tell, there is only a single method and I don't think it needs much more coverage than this single spec for now.

### CODE REVIEW

**Comment by pirog** [comment]
@dustinleblanc yeah this USED to be much more complicated in kalabox before we moved to to yargs and simplified the task structure. 

+1 for validation but we probably want to do fairly weak validation eg check for the minimal set of required props for a task. Do you want to do that as part of this PR or an additional task?

**Comment by dustinleblanc** [comment]
Yeah I can probably add a validator

**Comment by pirog** [comment]
I think the minimal set for validation here are just `name`, `command`, `description` and `run`

**Comment by dustinleblanc** [comment]
@pirog I am still working on the validation, decided to check this out: https://github.com/hapijs/joi

Seems like the most straight-forward way for us to do object validation project wide. Any objections to using this for the validation?

**Comment by pirog** [comment]
@dustinleblanc i was thinking something a little less involved eg just checking to see if the task object has the given keys or something but i guess we can do this too.

**Comment by dustinleblanc** [comment]
I'm not married to the deeper validation, if your spidey sense says err on the side of basic I can just check that the keys exist rather than validate the types of each key and marking some required, etc

**Comment by pirog** [comment]
@dustinleblanc yeah i mean we've NEVER had an issue with task validation before so i dont think we need to do something that intense. Check for keys, if not there maybe throw an error with a link to the 
https://docs.devwithlando.io/api/api.html#landotasksadd or something like that.

Probably is fine to build it directly into the add function itself

**Comment by dustinleblanc** [comment]
Validation simplified right into function, spec added, should be ready for review/merge

**Comment by pirog** [comment]
@dustinleblanc very nice, always good to adhere to the glory of YAGNI

File: lib/tasks.js [hunk 1]
```diff
@@ -58,12 +58,12 @@ exports.tasks = [];
...
-
-  // Build a task object
-  var tasker = _.merge(task, {name: name});
-
-  // Add to the task
-  this.tasks.push(tasker);
+exports.add = (name, task) => {
+  const mergedTask = _.merge(task, {name: name});
+  // Minimal task validation, just checking that common properties exist. If not, throw error.
+  if (!(_.has(mergedTask, 'name') && _.has(mergedTask, 'command') && _.has(mergedTask, 'describe') && _.has(mergedTask, 'run'))) {
```
Comment (pirog) [inline lib/tasks.js]: @dustinleblanc not a block to merge but there's gotta be a ... shorter way to do this? 

```js
const requiredKeys = ['name', ... 'whatever'];
if (!_.isEmpty(_.difference(_.intersection(requiredKeys, _.keys(mergedTask), requiredKeys))
```
but better

Comment (dustinleblanc) [reply lib/tasks.js]: Yeah I almost threw it into its own function, also wanted to method chain like
```js
_(mergedTask).has('foo').has('bar)
```
I'd actually prefer:

```js
if(!validTask(task))
```
And just do the dirty elsewhere...

Comment (dustinleblanc) [reply lib/tasks.js]: Which actually, ruby has a convention of suffixing boolean functions with `?`

I'd absolutely love:
```js
unless validTask?(task) 
  throw a Fit()
```

**Pull request merged**